src/vcf/record.cpp

```diff
@@ -330,9 +335,18 @@ namespace ebi
                                            "Allele index " + std::to_string(num_allele)
                                                    + " is greater than the maximum allowed "
                                                    + std::to_string(alternate_alleles.size()),
-                                           "GT"};
+                                           "GT", ploidy};
             }
         }
+
+        if (alleles.size() != ploidy) {
+            std::stringstream ss;
+            ss << "The specified ploidy for contig \"" << chromosome << "\" was " << ploidy
+               << ", which doesn't match the genotype \"";
+            util::print_container(ss, alleles, "", "/", "");
+            ss << "\" which has ploidy " << alleles.size();
+            throw new SamplesFieldBodyError{line, ss.str(), "GT", ploidy};
```

Comment:
The fixer takes care of that. It only removes non-GT fields, in GT fills with `.`s.

IMO the error is the same in both cases: a field in the samples format is incorrect. Furthermore, if we wanted to fix it in several ways, we should add another `Fixer` for the same error.